### PR TITLE
Step 1170 envs

### PR DESCRIPTION
**Pull request by godrei**
### Checklist

- [x] I've read and accepted the [Contribution Guidelines](https://github.com/bitrise-steplib/.github/blob/main/CONTRIBUTING.md)
- Requires _NO_ [version update](https://semver.org/)

### Context

Introduce `env.Repository` to unify the step's environment handling (merge `envValueClearer`, `envmanEnvRepository` and  `osEnvRepository`).

 

### Changes

- Introduce and wire in `env.Repository`
- sshagent.AddKey got a new parameter (`socket`), which gets passed to the ssh-add command through the `SSH_AUTH_SOCK`, this way we can get rid of an unnecessary os.SetEnv
- `CombinedEnvValueClearer` and it's `UnsetByValue` implemented inline

### CODE REVIEW

File: env/env.go [hunk 1]
```diff
@@ -8,6 +8,51 @@ import (
 
 // TODO: Move to `go-utils`
 
+// Repository ...
+type Repository interface {
+	Set(key, value string) error
```
Comment (mateherber) [inline env/env.go:13]: Shouldn't we have a Get() too?

Comment (mateherber) [reply env/env.go:13]: Discussed

File: env/env.go [hunk 2]
```diff
@@ -8,6 +8,51 @@ import (
 
 // TODO: Move to `go-utils`
 
+// Repository ...
+type Repository interface {
```
Comment (Bence1001) [inline env/env.go:12]: I worry this name is still too ambiguous. Also, a variable that refers to an instance is named `envRepository` and not just `repository`, so I would expect the interface to be named `EnvRepository` as well (or something similar).

Wdyt?

**Review by Bence1001 — APPROVED** [review]
LGTM, I have one comment regarding naming.

File: env/env.go (hunk 2 above)
Comment (godrei) [reply env/env.go:12]: golint fails for `env.EnvRepository`: `type name will be used as env.EnvRepository by other packages and that stutters; consider calling this Repository`

`Repository` in itself would be ambiguous, but you refer to it as `env.Repository`, which is not imo. When you assign it to a variable you can use `envRepository` as var name.

**Pull request merged**